fiaas_deploy_daemon/deployer/kubernetes/adapter.py

```diff
@@ -29,19 +29,22 @@ class K8s(object):
     """Adapt from an AppSpec to the necessary definitions for a kubernetes cluster
     """
 
-    def __init__(self, config, service_deployer, deployment_deployer, ingress_deployer, autoscaler):
+    def __init__(self, config, service_deployer, deployment_deployer, ingress_deployer, autoscaler, service_account_deployer):
```

Comment:
With this change you'll have to update some of the tests that pass in an argument here - I think if you look, you'll see how the test is creating mocks for the other parameters already, and you can follow the same pattern for this one. A test in that class that verifies the deployer is only called when the config says it's supposed to be would be good to see.